src/Models/FormSchema.cs

```diff
@@ -38,19 +38,20 @@ public FormSchema()
             EnvironmentAvailabilities = new List<EnvironmentAvailability>();
         }
 
-        public Page GetPage(string path)
+        public Page GetPage(IPageHelper pageHelper, string path)
         {
-            Page page;
             try
             {
-                page = Pages.SingleOrDefault(_ => _.PageSlug.ToLower().Trim() == path.ToLower().Trim());
+                var pages = Pages.Where(_ => _.PageSlug.ToLower().Trim() == path.ToLower().Trim()).OrderByDescending(_ => _.RenderConditions.Count).ToList();
+
+                var page = pageHelper.GetPageWithMatchingRenderConditions(pages);
+
+                return page;
```

Comment:
if there is only a single page, can we return straight away as there is no need to process any conditions also will reduce calls to the cache

Reply:
Good point, will fix shortly

Reply:
Fix pushed